### PR TITLE
add overflow checks and misc fixes

### PR DESCRIPTION
**Pull request by vchong**
@ruchi393 @jbech-linaro

### CODE REVIEW

File: keymaster/ta/parsel.c [hunk 1]
```diff
@@ -15,12 +15,29 @@
...
 
+#include <stdint.h>
 #include "parsel.h"
 #include "attestation.h"
 #include "generator.h"
+#include "util.h"
+
+bool TA_is_out_of_bounds(uint8_t *ptr, uint8_t *end, size_t size)
+{
+	uintptr_t res;
```
Comment (jockebech) [inline keymaster/ta/parsel.c]: Coding guidelines, initialize?

File: keymaster/ta/parsel.c [hunk 2]
```diff
@@ -64,13 +82,16 @@ int TA_deserialize_blob_akms(uint8_t *in, const uint8_t *end,
...
 		}
 		TEE_MemMove(data, in, blob->data_length);
-		in += blob->data_length;
 		blob->data = data;
 	} else {
 		/* Not allocate memory, it can be too large */
 		blob->data = in;
-		in += blob->data_length;
 	}
+	if (ADD_OVERFLOW((uintptr_t)in, blob->data_length, &sum)) {
```
Comment (jockebech) [inline keymaster/ta/parsel.c]: We use this directly instead of the overflow functions, since we don't know the end of sum so to say? Correct?

**Review by jockebech — COMMENTED** [review]
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

File: keymaster/ta/parsel.c (hunk 1 above)
Comment (vchong) [reply keymaster/ta/parsel.c]: Oops, missed that. Will fix. Thanks.

File: keymaster/ta/parsel.c (hunk 2 above)
Comment (vchong) [reply keymaster/ta/parsel.c]: Sorry, this change isn't really required as the steps are 1. check for out of bounds 2. add size to pointer, and/or use pointer and size as args in memcpy or TEE_MemMove(). 1 is already done in L71 (or 70) above so we can just add the size here without further checking.


Will remove this part.

**Comment by vchong** [comment]
Comments addressed and tag added.
Forced pushed since changes were minor.

@ruchi393 Please comment or merge as necessary.

**Comment by ruchi393** [comment]
`Acked-by: Ruchika Gupta <ruchika.gupta@linaro.org>`

**Comment by vchong** [comment]
Tag added and merged. Thanks!

**Pull request merged**